### PR TITLE
Update package:js version constraints to ^0.7.0 and update test versions to match

### PR DESCRIPTION
**Pull request by srujzs**
This removes the dependency on `dart:js` and allows that library to be removed from dart2wasm. Also updates `test` dependencies to work with the new `package:js` constraints.

### CODE REVIEW

**Review by eyebrowsoffire — APPROVED** [review]
LGTM!

**Comment by jason-simmons** [comment]
When I tried this locally I needed some additional updates:

In `lib/web_ui/pubspec.yaml`:
```
test: 1.25.8
matcher: 0.12.16+1
```

In  `web_sdk/web_engine_tester/pubspec.yaml`:
```
test: 1.25.8
```

**Comment by srujzs** [comment]
Ah, that makes sense, `package:test` didn't upgrade their `js` constraints to allow `0.7.0` until `1.25.1`: https://github.com/dart-lang/test/commit/fe3102eeb9b528083f2ce7405c78da24ae6fd0da.

I am little confused on the need to change `matcher` though.

**Comment by jason-simmons** [comment]
Yes - the new version of `matcher` is compatible with the `test_api` version used by the `test: 1.25.8`

(see https://github.com/dart-lang/matcher/commit/7ea90476559e30d97ec950d161a9b7968853bdbb)

**Comment by srujzs** [comment]
It looks like the PR in https://github.com/flutter/engine/pull/54626 removes the `matcher` dependency anyways.

**Comment by jason-simmons** [comment]
I think this needs to be rebased in order to pass CI

File: lib/web_ui/pubspec.yaml [hunk 1]
```diff
@@ -6,7 +6,7 @@ environment:
   sdk: '>=3.6.0-0 <4.0.0'
 
 dependencies:
-  js: 0.6.4
+  js: ^0.7.0
```
Comment (kevmoo) [inline lib/web_ui/pubspec.yaml:9]: I wonder if deprecations will start showing up then?

Comment (srujzs) [reply lib/web_ui/pubspec.yaml:9]: What deprecations are you referring to?

Comment (kevmoo) [reply lib/web_ui/pubspec.yaml:9]: Oh snap. I swore we marked libraries in pkg:js deprecated. Maybe not...

**Pull request merged**